algorithms.tex

```diff
@@ -34,18 +34,70 @@ \subsection{Numerics}
 -6.919482633683687653243407e-88
 \end{verbatim}
 
-The numerical evaluation works with complex numbers and supports
+The \texttt{evalf} method works with complex numbers and supports
 more complicated expressions, such as
-higher special functions, infinite series and integrals.
+special functions, infinite series and integrals.
+
+SymPy does not track the accuracy of
+approximate numbers outside of \texttt{evalf}.
+The familiar dangers of floating-point arithmetic apply, and 
+symbolic expressions containing floating-point numbers should be treated
+with some caution.
+This approach is similar to Maple and Maxima.
+
+By contrast, Mathematica uses a form
+of significance arithmetic~\cite{Sofroniou2005precise} for approximate numbers.
+This offers further protection against numerical errors,
+but leads to non-obvious semantics while
+still not being mathematically rigorous
+(for a critique of significance arithmetic, see Fateman~\cite{Fateman1992}).
+SymPy's \texttt{evalf} internals are non-rigorous in the same sense,
+but have no bearing on the semantics of floating-point
+numbers in the rest of the system.
+
+\subsubsection{Code generation}
+
+SymPy's \texttt{lambdify} can be used to convert a symbolic expression to a
+callable Python function for faster numerical evaluation.
+Various back ends are supported. The following example
+demonstrates creating a NumPy-based function from
+a SymPy expression, which automatically supports
+vectorized array evaluation [NumPy citation?]:
 
-[todo: contrast with mathematica's significance arithmetic? cite Sofroniou2005precise, maybe Fateman]
+\begin{verbatim}
+>>> f = lambdify((x, y), sin(x*y)**2, modules='numpy')
+>>> from numpy import array
+>>> f(array([1,2,3]), array([4,5,6]))
+array([ 0.57275002,  0.29595897,  0.56398184])
+\end{verbatim}
+
+SymPy can also generate C, C++, Fortran77, Fortran90 and
+Octave/Matlab source code, via the \texttt{codegen} function. [document this?]
```

Comment:
Code generation is mentioned in the table at https://github.com/sympy/sympy-paper/pull/42, which should be sufficient. 

The primary example in this section is `lambdify`, which is technically code generation, but probably not what first comes to mind for most people. A better goal for this section might be interaction with other Python libraries (particularly numpy).